toolkit/scripts/precache.mk

```
@@ -0,0 +1,50 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+# Contains:
+#	- Tools to fully hydrate the rpm cache from upstream repos with the any versions that match local .spec files
+
+.PHONY: pre-cache clean-precache
+
+precache_state_dir = $(CACHED_RPMS_DIR)/precache
+precache_downloaded_files = $(precache_state_dir)/downloaded_files.txt
+precache_chroot_dir = $(precache_state_dir)/chroot
+
+$(call create_folder,$(precache_state_dir))
+
+clean-cache: clean-precache
+clean: clean-precache
+clean-precache:
+	@echo Verifying no mountpoints present in $(precache_chroot_dir)
+	$(SCRIPTS_DIR)/safeunmount.sh "$(precache_chroot_dir)" && \
+	rm -rf $(precache_chroot_dir)
```

If we're running `clean`, we don't want to leave any lingering cache behind. I'd update `clean-precache`:

```suggestion
	rm -rf $(precache_state_dir)
```

and if we want to have an option to keep the cache, we can create a separate `clean-precache-chroot` target or something along these lines.